src/Microsoft.Diagnostics.ExtensionCommands/VerifyObjectCommand.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = "verifyobj", Help = "Checks the given object for signs of corruption.")]
+    public sealed class VerifyObjectCommand : CommandBase
+    {
+        [ServiceImport]
+        public ClrRuntime Runtime { get; set; }
+
+        [ServiceImport]
+        public IMemoryService Memory { get; set; }
+
+        [Argument(Name = "ObjectAddress", Help = "The object to verify.")]
+        public string ObjectAddress { get; set; }
+
+        public override void Invoke()
+        {
+            if (!TryParseAddress(ObjectAddress, out ulong objAddress))
+            {
+                throw new ArgumentException($"Invalid object address: '{ObjectAddress}'", nameof(ObjectAddress));
+            }
+
+            bool isNotCorrupt = Runtime.Heap.FullyVerifyObject(objAddress, out IEnumerable<ObjectCorruption> corruptionEnum);
+            if (isNotCorrupt)
+            {
+                Console.WriteLine($"object {objAddress:x} is a valid object");
```

Comment:
```suggestion
                Console.WriteLine($"object 0x{objAddress:x} is a valid object");
```
I'd prefer the `0x`-prefix to denote hexadecimal numbers.
If it's in this context very obvious, so please ignore that comment. But within .NET (especially config environment variables) there's some mismatch between decimal and hex, so such a prefix helps when reading values for someone who uses this from time to time only.

Reply:
Thanks!  I committed these changes.  I actually play to go back through all of the C# based implementations and make them consistent, improve DML support, and a few other cleanups.  It's been on my roadmap but I edited it to make it more explict:  "Improve DML handling, improve TableOutput, make output consistent especially with respect to hex/non hex values".  (#3736)

I plan to be done with C++ -> C# rewrites next week (hopefully), at which point I'm working on fit and finish things like this across all of the ExtensionCommands library.